app/src/main/java/com/example/sharingang/ui/fragments/DetailedItemFragment.kt

```diff
@@ -223,19 +223,19 @@ class DetailedItemFragment : Fragment() {
 
     private fun buyItem() {
         val quantity: Int = binding.quantity?.toIntOrNull() ?: 1
-        if (quantity > args.item.quantity || quantity < 1) {
+        val boughtItem = args.item
+        if (quantity > boughtItem.quantity || quantity < 1) {
             Toast.makeText(context, getString(R.string.incorrect_quantity), Toast.LENGTH_SHORT).show()
-            binding.buyButton.isEnabled = true
         } else {
             lifecycleScope.launch {
-                if (paymentProvider.requestPayment(args.item, quantity)) {
-                    itemViewModel.sellItem(args.item)
-                    val newQuantity = args.item.quantity - quantity
+                binding.buyButton.isEnabled = true
+                if (paymentProvider.requestPayment(boughtItem, quantity)) {
+                    itemViewModel.sellItem(boughtItem)
```

Comment:
I think you should only set it to sold after the new quantity has been updated.

Reply:
Yeah turns out I didn't need that function in the first place so I just removed it, it was from old code.